src/storage-types/src/controller.rs

```diff
@@ -287,6 +290,7 @@ impl fmt::Display for StorageError {
             Self::InvalidUsage(err) => write!(f, "invalid usage: {}", err),
             Self::ResourceExhausted(rsc) => write!(f, "{rsc} is exhausted"),
             Self::ShuttingDown(cmp) => write!(f, "{cmp} is shutting down"),
+            Self::RaceDetected => write!(f, "detected a race, please retry"),
```

Comment:
Is there a way to return an http retry code for this?

Reply:
Refactored this away